lib/cob_index.rb

```diff
@@ -2,14 +2,29 @@
 
 require "cob_index/version"
 require "cob_index/dot_properties"
+require "cob_index/nokogiri_indexer"
 require "traject"
 
+
 module CobIndex
   module CLI
-    def self.ingest(commit: false)
+    def self.ingest(commit: false, marc_xml: "")
       indexer = Traject::Indexer::MarcIndexer.new("solr_writer.commit_on_close": commit)
       indexer.load_config_file("#{File.dirname(__FILE__)}/cob_index/indexer_config.rb")
-      indexer.process(StringIO.new(open(ARGV[0]).read))
+      indexer.process(StringIO.new(marc_xml))
+    end
+
+
+    def self.delete(commit: false, xml: "")
+      settings =  {
+        "nokogiri.each_record_xpath" => "/oai:OAI-PMH/oai:ListRecords/oai:record",
+        "nokogiri.namespaces" => { "oai" => "http://www.openarchives.org/OAI/2.0/" },
+        "solr_writer.commit_on_close" => commit
```

Comment:
@dkinzer Will this just end up being one commit at the end of the task?

Reply:
No, the default is not to commit so here commit = false.

That  is there so we can pass `cob_index delete --commit file` which is useful for local development purposes.